crates/publish/src/oci/client.rs

```diff
@@ -209,28 +214,86 @@ impl Client {
         ))
     }
 
+    /// Save a credential set containing the registry username and password.
+    pub async fn login(
+        server: impl AsRef<str>,
+        username: impl AsRef<str>,
+        password: impl AsRef<str>,
+    ) -> Result<()> {
+        // We want to allow a user to login to both https://ghcr.io and ghcr.io.
+        let server = match server.as_ref().parse::<Url>() {
+            Ok(url) => url.host_str().unwrap_or(server.as_ref()).to_string(),
+            Err(_) => server.as_ref().to_string(),
+        };
+
+        // First, test the credentials. If a user accidentally enters a wrong credential set, this
+        // can catch the issue early rather than the user having to figure it out on their own.
+        Self::test_credentials(&server, &username, &password)
+            .await
+            .context("cannot check credentials")?;
+
+        // Save an encoded representation of the credential set in the local configuration file.
+        let mut auth = AuthConfig::load_default().await?;
+        auth.insert(server, username, password)?;
```

Comment:
I know this is "how its done" but WHAT YEAR IS THIS?!?!

Reply:
ref #940

Reply:
I'm mostly complaining about storing _passwords_ rather than tokens, but thats a good point too.

Reply:
Yeah, we need to address both of them..